src/engine/tt_hash_table.h

```diff
@@ -14,20 +15,87 @@ enum TtHashFlag : uint8_t {
     TtHashBeta = 2,
 };
 
+/*
+ * Non-key members are encoded for compression. From lower bits to higher, entries are:
+ * depth: 7 bits, i.e. 0000 ... 0000 0000 0111 1111
+ * flag: 3 bits, i.e. 0000 ... 0000 0011 1000 0000
+ * score: 17 bits (see board_defs.h)
+ * generation: 8 bits
+ * move: 26 bits (see move_types.h)
+ * */
+
+class TtHashEntryData {
+public:
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshift-count-overflow"
+    explicit TtHashEntryData(uint64_t depth = 0, uint64_t flag = TtHashExact, int64_t score = 0, uint64_t generation = 0, uint64_t move = {})
```

Comment:
No need for default args when data itself is default initialized. Probably the issue you tried to solve here?:)
```suggestion
    explicit TtHashEntryData(uint8_t depth = 0, uint8_t flag = TtHashExact, int32_t score = 0, uint8_t generation = 0, uint32_t move = {})
```

Reply:
Nice spot, changed.